src/api.js

```diff
@@ -272,3 +278,44 @@ api.declare({
     }),
   });
 });
+
+api.declare({
+  name: 'isInstalledFor',
+  title: 'Check if Repository has Integration',
+  description: [
+    'Checks if the integration has been installed for',
+    'a given repository of a given organization or user.',
+  ].join('\n'),
+  stability: 'experimental',
+  method: 'get',
+  route: '/repository/:owner/:repo',
+  output: 'is-installed-for.json',
+}, async function(req, res) {
+  // Extract owner and repo from request into variables
+  let {owner, repo} = req.params;
+
+  // Look up the installation ID in Azure. If no such owner in the table, no error thrown
+  let ownerInfo = await this.OwnersDirectory.load({owner}, true);
+
+  if (ownerInfo) {
+    let instGithub = await this.github.getInstallationGithub(ownerInfo.installationId);
+    let reposList = await instGithub.integrations.getInstallationRepositories();
+
+    // GitHub API returns an array of objects, each of wich has an array of repos
+    // [ { repositories: [ [Object], [Object], [Object] ] } ]
+    console.log('reposList', reposList);
+    try {
+      let installed = reposList.reduce((a, b) => a.repositories.concat(b.repositories), {repositories: []})
```

Comment:
When `reposList` is `[]`, this does something different than when it's actually a list of repositories.  That worries me, even though it's unlikely to happen in real life.  That's why you're getting "reposList.reduce(...).map is not a function", I think.

What about this:

```js
reposList.reduce((a, b) => ({repositories: a.repositories.concat(b.repositories)}), {repositories: []})
  .repositories
  .map(repo => repo.name)
  .includes(repo);
```

This just makes the reducer function return `{repositories: [..]}` instead of returning `[..]` directly.  Done this way, its inputs and output all have the same shape (`{repositories: [..]}`), so whether we have an empty `reposList`, and thus return the initial value, or do a bunch of reductions, the shape always stays the same.  In general, this is a good principle when using reduce.